modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1024,24 +1024,81 @@ public final HitInfo hitTest(Point2D point) {
         double x = point.getX() - getX();
         double y = point.getY() - getY() + getYRendering();
         GlyphList[] runs = getRuns();
-        int runIndex = 0;
+        int runIndex = findRunIndex(x, y, runs);
+
+        int textRunStart = 0;
+        int curRunStart = 0;
         if (runs.length != 0) {
+            textRunStart = findFirstRunStart(x, y, runs);
+            curRunStart = ((TextRun) runs[runIndex]).getStart();
+        }
+        TextLayout.Hit h = layout.getHitInfo((float)x, (float)y, getText(), textRunStart, curRunStart, false);
+        return new HitInfo(h.getCharIndex(), h.getInsertionIndex(), h.isLeading());
+    }
+
+    private int findFirstRunStart(double x, double y, GlyphList[] runs) {
+        int start = Integer.MAX_VALUE;
+        for (GlyphList r: runs) {
+            if (((TextRun) r).getStart() < start) {
+                start = ((TextRun) r).getStart();
+            }
+        }
+        return start;
+    }
+
+    private int findRunIndex(double x, double y, GlyphList[] runs) {
+        int ix = 0;
```

Comment:
The naming leaves something to be desired.  `ix` is the run index is it not?  Maybe leave it as `runIndex` ?